### PR TITLE
test: add recovery test for background_ddl

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Later we can expand it to randomly issue background ddl for mview statements.

See: https://github.com/risingwavelabs/risingwave/pull/13552

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/tests/simulation/src/slt.rs [hunk 1]
```diff
@@ -59,16 +70,31 @@ impl SqlCmd {
...
-        .split_once(' ')
-        .unwrap_or_default()
-        .0
-        .to_lowercase();
-    match cmd.as_str() {
-        "create" => SqlCmd::Create {
-            is_create_table_as: is_create_table_as(sql),
+    let mut tokens = sql.split_whitespace().map(|s| s.to_lowercase());
+    fn next_token(tokens: &mut impl Iterator<Item = String>) -> String {
+        tokens.next().unwrap()
```
Comment (BugenZhao) [inline src/tests/simulation/src/slt.rs:75]: If we have some tests intended to fail with parse error, will this lead to panic of the test driver?

File: src/tests/simulation/src/slt.rs [hunk 2]
```diff
@@ -59,16 +70,31 @@ impl SqlCmd {
...
+            },
         },
+        "set" => {
+            if sql.contains("background_ddl") {
+                let enable = sql.contains("true");
+                SqlCmd::SetBackgroundDdl { enable }
+            } else {
+                SqlCmd::Others
+            }
+        }
```
Comment (BugenZhao) [inline src/tests/simulation/src/slt.rs:97]: As `SET` will be replayed after recovery, is it okay to directly fetch the value of `background_ddl` with SQL?

File: src/tests/simulation/src/slt.rs [hunk 3]
```diff
@@ -189,14 +219,39 @@ pub async fn run_slt_task(cluster: Arc<Cluster>, glob: &str, opts: &KillOpts) {
...
+                            // wait for background ddl to finish and succeed.
+                            let rw = RisingWave::connect("frontend".into(), "dev".into())
+                                .await
+                                .unwrap();
+                            let client = rw.pg_client();
+                            if client.simple_query("WAIT;").await.is_ok()
+                                && client
+                                    .query(
+                                        "select count(*) from pg_matviews where matviewname=$1;",
+                                        &[&name],
```
Comment (BugenZhao) [inline src/tests/simulation/src/slt.rs:235]: Should we test against the count?

**Comment by kwannoel** [comment]
Migrating to https://github.com/risingwavelabs/risingwave/pull/13552.

@BugenZhao I will address your comments there.

**Pull request closed without merging**